protocol/x/vault/keeper/orders.go

```diff
@@ -344,6 +367,26 @@ func (k Keeper) GetVaultClobOrders(
 		orders[2*i+1] = constructOrder(clobtypes.Order_SIDE_BUY, i, orderIds[2*i+1])
 	}
 
+	if vaultParams.Status == types.VaultStatus_VAULT_STATUS_CLOSE_ONLY {
+		// In close-only mode with non-zero leverage.
+		reduceOnlyOrders := make([]*clobtypes.Order, quotingParams.Layers)
+		reduceOnlyMaxOrderSize := k.GetVaultInventoryInPerpetual(ctx, vaultId, perpetual.Params.Id)
+		stepSize := lib.BigU(clobPair.StepBaseQuantums)
+		reduceOnlyMaxOrderSize.Quo(reduceOnlyMaxOrderSize, stepSize)
+		reduceOnlyMaxOrderSize.Mul(reduceOnlyMaxOrderSize, stepSize)
+		// If vault is long, only need sell orders (even indices).
+		i := 0
+		if leverage.Sign() < 0 {
+			// If vault is short, only need buy orders (odd indices).
+			i = 1
+		}
+		for ; i < len(orders); i += 2 {
+			reduceOnlyOrders[i/2] = orders[i]
```

Comment:
nit: Rather than do the division, maybe just keep another index counter that you increment per loop iteration?

Reply:
see below comment